### PR TITLE
Update SDK

### PR DESCRIPTION
**Pull request by halter73**

### CODE REVIEW

**Comment by halter73** [comment]
@jjonescz @sharwell @333fred @chsienki After updating the .NET SDK from 8.0.100-preview.5.23257.1 to 8.0.100-preview.5.23265.3, we're seeing errors complaining about `async` without `await` in the razor source generator.

```
/mnt/vss/_work/1/s/src/Identity/UI/src/Microsoft.NET.Sdk.Razor.SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator/Areas_Identity_Pages_V5_Account_Manage__Layout_cshtml.g.cs(102,46): error CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/mnt/vss/_work/1/s/src/Identity/UI/src/Microsoft.AspNetCore.Identity.UI.csproj]
```

Could this be related to https://github.com/dotnet/razor/pull/8346 which was merged last week and was intended to suppress these warnings?

**Comment by jjonescz** [comment]
@halter73 You're right, that bug fix was wrong in some cases. I opened a follow up to fix this: https://github.com/dotnet/razor/pull/8709

**Comment by 333fred** [comment]
https://github.com/dotnet/razor/pull/8719 was merged, reverting the change here.

**Comment by wtgodbe** [comment]
https://github.com/dotnet/sdk/pull/32650 brings that change into the SDK. Once that goes in & sdk -> installer flow is done we can grab the new SDK

**Comment by wtgodbe** [comment]
@eerhardt we're starting to see trimming test failures here

**Comment by eerhardt** [comment]
I think this looks very similar to https://github.com/dotnet/sdk/issues/32272.

FYI - @nagilson and @sbomer 

We could try switching:

https://github.com/dotnet/aspnetcore/blob/7c89649e1a4d5ab3a08ee317a8faeb81d59def57/eng/testing/linker/SupportFiles/Directory.Build.props#L8

to be `<SelfContained>true</SelfContained>`.

But honestly - I'm not sure I understand the benefit of `PublishSelfContained` if this scenario doesn't work. FYI - @baronfel @richlander @dsplaisted

**Comment by nagilson** [comment]
> @eerhardt 
> But honestly - I'm not sure I understand the benefit of `PublishSelfContained` if this scenario doesn't work. FYI

https://github.com/dotnet/sdk/issues/32277#issuecomment-1550369783 This comment should explain the feature a bit better, and I agree with your suggestion.

File: src/Tools/Extensions.ApiDescription.Server/src/build/Microsoft.Extensions.ApiDescription.Server.targets [hunk 1]
```diff
@@ -57,7 +57,7 @@
       <_Command Condition=" '$(ProjectAssetsFile)' != '' ">$(_Command) --assets-file "$(ProjectAssetsFile)"</_Command>
       <_Command Condition=" '$(PlatformTarget)' != '' ">$(_Command) --platform "$(PlatformTarget)"</_Command>
       <_Command Condition=" '$(PlatformTarget)' == '' AND '$(Platform)' != '' ">$(_Command) --platform "$(Platform)"</_Command>
-      <_Command Condition=" '$(RuntimeIdentifier)' != '' ">$(_Command) --runtime "$(RuntimeIdentifier)"</_Command>
+      <_Command Condition=" '$(RuntimeIdentifier)' != '' ">$(_Command) --runtime "$(RuntimeIdentifier) --self-contained"</_Command>
```
Comment (BrennanConroy) [inline src/Tools/Extensions.ApiDescription.Server/src/build/Microsoft.Extensions.ApiDescription.Server.targets:60]: I don't know anything about this project. @captainsafia is this ok?

**Comment by BrennanConroy** [comment]
Had to react to the sdk breaking change: https://github.com/dotnet/docs/issues/33726

**Review by Tratcher — APPROVED** [review]

**Review by eerhardt — APPROVED** [review]

**Comment by mitchdenny** [comment]
These checks are green but I'm unsure if this is being held up by anything else? @halter73?

File: src/Tools/Extensions.ApiDescription.Server/src/build/Microsoft.Extensions.ApiDescription.Server.targets (hunk 1 above)
Comment (captainsafia) [reply src/Tools/Extensions.ApiDescription.Server/src/build/Microsoft.Extensions.ApiDescription.Server.targets:60]: Yes, this looks correct based on the [breaking change details](https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/6.0/runtimeidentifier-self-contained).

**Review by captainsafia — APPROVED** [review]

**Review by captainsafia — APPROVED** [review]

**Pull request merged**